accepted/0000-plug-an-play.md

```diff
@@ -0,0 +1,302 @@
+- Start Date: 13 Sep 18
+- RFC PR: https://github.com/yarnpkg/rfcs/pull/101
+- Yarn Issue: https://github.com/yarnpkg/yarn/issues/6382
+- Champion: Maël Nison (@arcanis - [Twitter](https://twitter.com/arcanis))
+
+# Plug'n'Play Whitepaper
+
+## 1. Summary
+
+We propose in this RFC a new alternative and entirely optional way to resolve dependencies installed on the disk, in order to solve issues caused by the incomplete knowledge Node has regarding the dependency tree. We also detail the actual implementation we went with, describing the rational behind the design choice we made.
+
+I'll keep it short in this summary since the document is already large, but here are some highlights:
+
+* Installs ran using Plug'n'Play are up to 70% faster than regular ones ([sample app](https://github.com/yarnpkg/pnp-sample-app))
+* Starting from this PR, Yarn will now be on the path to make yarn install a no-op on CI
+* Yarn will now be able to tell you precisely when you forgot to list packages in your dependencies
+* Your applications will boot faster through a hybrid approach of static resolutions
+
+This is but a high-level description of some of the benefits unlocked by Plug'n'Play, I encourage you to give a look at the document for more information about the specific design choices - and in case anything is missing, please ask and I'll do my best to explain them more in depth!
+
+## 2. Motivation
+
+When the first Javascript package managers appeared, they had to use the tools available to them. As a consequence, they were implemented on top of the Node resolution algorithm, copying the packages in such a layout that Node would be able to find them without external tool. This allowed the ecosystem to thrive, which interestingly revealed some of the scalability flaws in this approach:
+
+* The Node resolution algorithm isn't aware of what packages are, and as a result doesn't know anything about dependencies either. When a module makes a require call, Node simply traverses the filesystem until it finds something that matches the request, and uses it.
+* Installations take time, and the time required to copy files from the package managers' caches to the `node_modules` folders is one of the main bottleneck as it is heavily I/O bound, which cannot be easily optimized. Copy-on-Write filesystems can alleviate this issue to an extent but come with their own set of drawbacks, such as a lack of global support.

```

Comment:
typo: managers' => managers

Reply:
This is not a typo. This is a use of the apostrophe to indicate a plural possessive, i.e. "the caches of the package managers".

(While I'm talking about this anyway: "without external tool" in line 23 should be "without external tools", or perhaps "without an external tool".)